source/ui/dataman/AMSpectrumAndPeriodicTableView.cpp

```diff
@@ -0,0 +1,388 @@
+#include "AMSpectrumAndPeriodicTableView.h"
+
+#include "dataman/AMScan.h"
+#include "ui/dataman/AMColoredTextToolButton.h"
+#include "ui/util/AMPeriodicTableDialog.h"
+
+#include "MPlot/MPlotTools.h"
+
+#include <QStringBuilder>
+
+ AMSpectrumAndPeriodicTableView::~AMSpectrumAndPeriodicTableView(){}
+AMSpectrumAndPeriodicTableView::AMSpectrumAndPeriodicTableView(QWidget *parent)
+	: QWidget(parent)
+{
+	emissionLineValidator_ = new AMNameAndRangeValidator(this);
+	pileUpPeakValidator_ = new AMNameAndRangeValidator(this);
+	combinationPileUpPeakValidator_ = new AMNameAndRangeValidator(this);
+
+	periodicTable_ = new AMSelectablePeriodicTable(this);
```

Comment:
Because there are virtual functions which build the objects and virtual functions are not properly referenced to their appropriate class until after the object has been constructed.